### DISCUSSION TITLE
Add status column to Spree::Refund

### DISCUSSION THREAD

**Discussion by abbyhudson13** [solidusio/solidus#4862] [New Features or Ideas]
As discussed in the [slack channel](https://solidusio.slack.com/archives/C0JBKDF35/p1674035519730229), `charge.refunded` events from buy now pay later payment methods eg. Klarna come through from Stripe with a `pending` status. Currently we only create a refund in Solidus when the status is `succeeded`. The only solution would be to wait for the `refund.updated` event to come through from Stripe when the refund has succeeded and create the refund in Solidus following this event. However, this is confusing because there is no indication within Solidus that the refund request has been sent until this event comes through. 

The suggestion would be to create a `status` column on the Spree::Refund (as @AlistairNorman has said is in an unreleased extension already), so the refund would be created with a `pending` status until the succeeded status comes back from Stripe, at which point the status of the refund also changes to `succeeded`.

**Comment by seandawes** [comment]
We are not up to current version of core and have modifications to the states but one comment here I would have would be to follow similar state logic found in other areas and their naming conventions. So table spree_refunds would get spree_refunds.state of which naming could be the following

- pending
- complete
- canceled (would there be an event where this line needs a canceled state?)

I think one of the biggest hurdles for anyone new including myself was figuring out what was what and the slight variants at times in naming conventions. We have build external apps which work with solidus and I actually follow similar naming conventions and logic outside so that it makes more sense when working with our internal data

**Reply by abbyhudson13** [reply]
I think the status of the refund from Stripe would come back as either `succeeded`, `pending` or `failed`
